ui/src/components/TaskDetailsCard/index.jsx

```diff
@@ -99,45 +105,63 @@ export default class TaskDetailsCard extends Component {
      * A GraphQL task response.
      */
     task: task.isRequired,
-    /**
-     * A collection of GraphQL task responses associated with the given task.
-     */
-    dependentTasks: arrayOf(
-      shape({
-        taskId: string.isRequired,
-        // note that status and metadata may be missing
-        status: shape({
-          state: string,
-        }),
-        metadata: shape({
-          name: string,
-        }),
-      })
-    ),
     dependents: shape({
       edges: arrayOf(task),
       pageInfo,
     }),
     onDependentsPageChange: func.isRequired,
+    user: object,
   };
 
   state = {
     showPayload: false,
+    dependentTasks: null,
+    loading: false,
   };
 
   handleTogglePayload = () => {
     this.setState({ showPayload: !this.state.showPayload });
   };
 
+  componentDidMount() {
+    this.fetchDependentTasks();
+  }
+
+  async fetchDependentTasks() {
+    const { task, user } = this.props;
+
+    this.setState({ loading: true });
+
+    if (task.dependencies.length > 0) {
```

Comment:
Will `dependencies` _always_ be defined here? I'm just wondering if we need to add optional chaining

Reply:
so `task` is coming from the graphql response, which includes this.. but I guess we can check additionally 👍